### ISSUE TITLE
Support Gradle build cache

### ISSUE THREAD

**Issue by 3flex** [Kotlin/dokka#453]
Supporting the build cache will help drastically reduce build times when the inputs to the dokka task haven't changed and the build is being done for the first time on a new machine.

This is particularly helpful in reducing CI build times.

https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_details

**Comment by rnett** [comment]
With the new plugin, I get:
```
> Task :core:dokkaHtmlMultiModule
Execution optimizations have been disabled for task ':core:dokkaHtmlMultiModule' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/core/krosstalk-base/build/dokka/htmlPartial'. Reason: Task ':core:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/core/krosstalk-client/build/dokka/htmlPartial'. Reason: Task ':core:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/core/krosstalk-server/build/dokka/htmlPartial'. Reason: Task ':core:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/core/krosstalk/build/dokka/htmlPartial'. Reason: Task ':core:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```
and
```
> Task :plugins:dokkaHtmlMultiModule
Execution optimizations have been disabled for task ':plugins:dokkaHtmlMultiModule' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/krosstalk-kotlinx-serialization/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-client/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-client/krosstalk-ktor-client-auth/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-client/krosstalk-ktor-client/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-server/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-server/krosstalk-ktor-server-auth-jwt/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-server/krosstalk-ktor-server-auth/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/home/runner/work/krosstalk/krosstalk/plugins/ktor-server/krosstalk-ktor-server/build/dokka/htmlPartial'. Reason: Task ':plugins:dokkaHtmlMultiModule' uses this output of task ':dokkaHtmlMultiModule' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

**Comment by rnett** [comment]
Not quite sure how you would do it, but it would be extremely nice if the cache was split so that if you change code without changing public declarations or documentation, the docs wouldn't be regenerated.

**Comment by keturn** [comment]
Has this been addressed by #1905?

**Comment by 3flex** [comment]
Thanks, looks like #1905 addressed this, closing.
